crates/noirc_frontend/src/hir/type_check/expr.rs

```diff
@@ -975,8 +981,14 @@ impl<'interner> TypeChecker<'interner> {
                         span,
                     });
                 }
-                let comptime = comptime_x.and(comptime_y, op.location.span);
-                Ok(Integer(comptime, *sign_x, *bit_width_x))
+                if op.is_bit_shift()
+                    && (*sign_x != Signedness::Unsigned || *sign_y != Signedness::Unsigned)
```

Comment:
Since there are merge conflicts to fix anyway, why are we checking if it `!=` unsigned rather than `== signed`? I'd think the later would be more intuitive.

Reply:
Done